src/category_theory/category/pairwise.lean

```diff
@@ -0,0 +1,150 @@
+import topology.sheaves.sheaf
+import category_theory.limits.preserves.basic
+
+/-!
+# The category of "pairwise intersections".
+
+Given `ι : Type v`, we build the diagram category `pairwise ι`
+with objects `single i` and `pair i j`, for `i j : ι`,
+whose only non-identity morphisms are
+`left : single i ⟶ pair i j` and `right : single j ⟶ pair i j`.
+
+We use this later in describing the sheaf condition.
+
+Given any function `U : ι → α`, where `α` is some complete lattice (e.g. `opens X`),
+we produce a functor `pairwise ι ⥤ αᵒᵖ` in the obvious way,
+and show that `supr U` provides a limit cone over this functor.
+-/
+
+noncomputable theory
+
+universes v u
+
+open topological_space
+open Top
+open opposite
+open category_theory
+open category_theory.limits
+
+namespace category_theory
+
+/--
+An inductive type representing either a single term of a type `ι`, or a pair of terms.
+We use this as the objects of a category to describe the sheaf condition.
+-/
+inductive pairwise (ι : Type v)
+| single : ι → pairwise
+| pair : ι → ι → pairwise
+
+variables {ι : Type v}
+
+namespace pairwise
+
+instance pairwise_inhabited [inhabited ι] : inhabited (pairwise ι) := ⟨single (default ι)⟩
+
+/--
+Morphisms in the category `pairwise ι`. The only non-identity morphisms are
+`left i j : single i ⟶ pair i j` and `right i j : single j ⟶ pair i j`.
+-/
+inductive hom : pairwise ι → pairwise ι → Type v
```

Comment:
Can you use the machinery on thin categories to derive the category instance form this definition?

Reply:
Unfortunately it's not a thin category, because there are two morphisms from `single i` to `pair i i`.

Reply:
Oh, of course :facepalm: